### PR TITLE
refactor(): support server+cli => ruins electron

### PR DESCRIPTION
**Pull request by ShaMan123**
* refactor into node app
* support http server: entry `server.mjs`
* support cli: entry `cli.mjs`, also as bin `deepnest ...`
* kill electron app: don't care about it but if anyone does then it is possible to revive it either by decoupling the node logic as I did in most places using `*.node.js` and reusing files in the electron app or by exposing the entire node app as a consumable under `main.js`. This will make it possible to bump the app to electron latest (33 at time of writing) because context isolation will be achieved by doing so.
* unknown bug when exporting to SVG, handled by a null check
* expose `index.js`
* svg parser fixes
* disable electron test
* update readme

### CODE REVIEW

**Comment by abebeos** [comment]
@ShaMan123 , i've done a quick 1.2.1 release (without much testing). you can at this point send a PR in for build/test changes&additions, see https://github.com/deepnest-io/Deepnest/pull/159#issuecomment-2525260696.

**Review by ShaMan123 — COMMENTED** [review]
cli and server up and running
electron app is down!

**Review by ShaMan123 — COMMENTED** [review]
Stable

File: main/svgparser.js [hunk 1]
```diff
@@ -1271,76 +1333,58 @@
...
-					if(command == 'Z' || command == 'z'){
-						x = x0;
-						y = y0;
-					}
-					p.pathSegList.appendItem(s);
-				}
+		const seglist = path.pathSegList || (path.pathSegList = new window.SVGPathSegList(path));
+		const indices = [];
+
+		for (let index = 0, x = 0, y = 0; index < seglist.numberOfItems; index++) {
```
Comment (ShaMan123) [inline main/svgparser.js:1344]: fix start

**Comment by Dexus** [comment]
@ShaMan123 any plans to build and convert to typescript? I'm currently working on a update to electron 33.2 which is done, but also on porting functions to rust and export them as nodejs bindings.

Maybe we can so make the code cleaner and have more options to run the nesting tools that anybody need.

**Comment by ShaMan123** [comment]
I needed this functionality for a server I am deploying.
That is done now, thankfully, and so am I.

The code quality of deepnest is extremely poor so the task was too demanding.
Whether it is whitespace, dead code, very old js or simply bad practice all around (endless data set mutation).

I like TS support and am not married to anything here apart for my need for it to be a module I can call and not an electron app (a very weird choice IMO).
I lack the time and interest to invest in deepnest anymore, though I appreciate the algorithm it wraps around.
You are welcome to any part of my fork.
I now understand the lifecycle of deepnest and it can be viewed in `index.mjs` with ease.
I decided to touch as less as possible with the notion of deepnest seeming to be dying out and since I don't want to diverge too much from upstream in case I need to maintain/update this fork in the future hoping for upstream fixes to some issues.
That is why I refrained from whitespace changes (prettier) and file renaming (`.ts`).

I have other forks I have to maintain because of dying open source projects. It is no fun.

**Comment by Dexus** [comment]
I know what you mean! Thank you for your great job and the current state.

**Pull request merged**